### ISSUE TITLE
QUESTION OF YOUR CODE

### ISSUE THREAD

**Issue by Sun-Rider** [nayeemrizve/ups#7]
Hi,

Thanks for the share. 

Based on your equation 3,

<img width="528" alt="Screen Shot 2021-07-06 at 9 42 09 am" src="https://user-images.githubusercontent.com/33716428/124525482-9b504600-de3e-11eb-8003-32d462a057d5.png">

The tilde y in here is the pre-generated hard pseudo label, and y hat is the model prediction. However, according to your code in [here](https://github.com/nayeemrizve/ups/blob/f3b49ace801a223e33e3b822a5137e20adce93dc/utils/train_util.py#L50-L51), 
``` python
            y_nl = torch.ones((nl_logits.shape)).to(device=args.device, dtype=logits.dtype)
            loss_nl += torch.mean((-torch.sum((y_nl * torch.log(pred_nl))*nl_mask, dim = -1))/(torch.sum(nl_mask, dim = -1) + 1e-7))
```

it looks like you only multiply the log(1- y hat) with the 1, instead of the (1-y tilde).

Do I miss something? 

Cheers,

**Comment by SydCS** [comment]
Same question. When y_tilde=0, y_bl=1 ?
